### PR TITLE
docs: how to write an extension and use the api

### PR DESCRIPTION
**Pull request by cdrage**
docs: how to write an extension and use the api

### What does this PR do?

Can never have too much documentation!

I started documenting my findings as I worked through all the podman desktop code.

This PR adds documentation for the following:
- Writing an extension
- Expanding the main Podman Desktop API with new functionality

### Screenshot/screencast of this PR

N/A

### What issues does this PR fix or reference?

N/A

### How to test this PR?

N/A

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -245,4 +245,135 @@ If you're unsure where to add code (renderer, UI, extensions, plugins) see the b
...
+Podman Desktop is organized so that you can modularly add new functionality in the form of "extensions" as well as the corresponding `extension-api`. This allows you to communicate with Podman Desktop without having to know the internal-workings. You look for the API call and Podman Desktop will do the rest.
+
+This is located in the `/extensions` folder.
+
+#### Creating a new extension
+
+When creating a new extension, import the extension API: `import * as extensionApi from '@tmpwip/extension-api';` All functionality with Podman Desktop is communicated through this API including registering the new extension. The API is located [here](https://github.com/containers/podman-desktop/blob/main/packages/extension-api/src/extension-api.d.ts).
+
+When loading an extension, Podman Desktop will:
+1. Search and load the `/src/extension.ts` file in the extension directory.
```
Comment (benoitf) [inline CONTRIBUTING.md]:
```suggestion
1. Search and load the javascript file specified in `main` entry of the `package.json` file in the extension directory.
```

**Comment by cdrage** [comment]
Updated! @benoitf

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -245,4 +245,135 @@ If you're unsure where to add code (renderer, UI, extensions, plugins) see the b
...
+Podman Desktop is organized so that you can modularly add new functionality in the form of "extensions" as well as the corresponding `extension-api`. This allows you to communicate with Podman Desktop without having to know the internal-workings. You look for the API call and Podman Desktop will do the rest.
+
+This is located in the `/extensions` folder.
+
+#### Creating a new extension
+
+When creating a new extension, import the extension API: `import * as extensionApi from '@tmpwip/extension-api';` All functionality with Podman Desktop is communicated through this API including registering the new extension. The API is located [here](https://github.com/containers/podman-desktop/blob/main/packages/extension-api/src/extension-api.d.ts).
+
+When loading an extension, Podman Desktop will:
+1. Search and load the JavaScript / TypeScript file specified in `main` entry of the `package.json` file in the extension directory (typically `extension.js`).
```
Comment (benoitf) [inline CONTRIBUTING.md]: it can't be TypeScript there, only JavaScript as it's the "result/output" that will be loaded by Node.js

**Review by benoitf — APPROVED** [review]
🎉

**Pull request merged**